mfhdf/hrepack/hrepacktst.c

```diff
@@ -110,29 +110,41 @@ static unsigned char *g_image_data = NULL;
  *
  *-------------------------------------------------------------------------
  */
-
 static int
 read_data(const char *fname)
 {
     int   i, n;
+    int   count;
     int   color_planes;
     char  str[20];
-    FILE *f;
+    FILE *f = NULL;
     int   w, h;
 
     f = fopen(fname, "r");
 
     if (f == NULL) {
         printf("Could not open file <%s>\n", fname);
-        return -1;
+        goto error;
     }
 
-    fscanf(f, "%s", str);
-    fscanf(f, "%d", &color_planes);
-    fscanf(f, "%s", str);
-    fscanf(f, "%d", &h);
-    fscanf(f, "%s", str);
-    fscanf(f, "%d", &w);
+    count = fscanf(f, "%s", str);
+    if (count != 1)
+        goto error;
+    count = fscanf(f, "%d", &color_planes);
+    if (count != 1)
+        goto error;
+    count = fscanf(f, "%s", str);
+    if (count != 1)
+        goto error;
+    count = fscanf(f, "%d", &h);
+    if (count != 1)
+        goto error;
+    count = fscanf(f, "%s", str);
+    if (count != 1)
+        goto error;
+    count = fscanf(f, "%d", &w);
+    if (count != 1)
+        goto error;
```

Comment:
fscanf will return EOF or the wrong number of matches if there were problems, so I just check for the correct number of matches.